adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/ADLSBatchSink.java

```diff
@@ -140,6 +166,12 @@ protected void validate() {
       if (!containsMacro("path") && !path.startsWith("adl://")) {
         throw new IllegalArgumentException("Path must start with adl:// for ADLS input files.");
       }
+      if (TEXT.equals(outputFormat) && textDelimiter == null) {
+        throw new IllegalArgumentException("Field delimiter must be specified for text format output files.");
+      }
+      if (AVRO.equals(outputFormat) && schema == null) {
+        throw new IllegalArgumentException("Output schema must be specified for avro format output files.");
+      }
```

Comment:
validate should also try and parse the schema if it is not null. You can just call getSchema(), which will do it.

Reply:
If schema is not null and the format is text, you should validate the field types to make sure only supported types are in the schema.